### PR TITLE
examples: cleanups for better user experience

### PR DESCRIPTION
**Pull request by EmbeddedAndroid**
Some notable changes:

  * Add argparse interface
  * Add hook for retrieving PyOTA library version
  * Reuse get_seed() to keep from displaying the seed in cleartext
  * Clean up IOTA library imports
  * Removed unused imports
  * Fix invalid depth
  * Misc PEP8 formatting fixes

### CODE REVIEW

**Comment by todofixthis** [comment]
Hey @EmbeddedAndroid thanks for submitting this PR!  I changed the target branch to `develop`.  This caused a couple of conflicts, as we're currently in the middle of PEP-8-ifying the codebase.

Once the conflicts are resolved, I think this is ready to merge 😺

**Comment by EmbeddedAndroid** [comment]
Thanks for reviewing, I'll get the conflicts sorted!

**Comment by EmbeddedAndroid** [comment]
Patches should apply cleanly to the develop branch now!

**Comment by todofixthis** [comment]
Sweet as.  Cheers for that!

Would you mind updating the formatting so that it matches PEP-8 (4-space indents, no spaces around `=` in function invocations, etc.)?

Everything else looks good!

**Comment by EmbeddedAndroid** [comment]
That should fix up the PEP-8 issues, let me know if you spot anything else. Cheers!

File: examples/address_generator.py [hunk 1]
```diff
@@ -15,7 +15,6 @@
 
 from iota import Iota, __version__
 from iota.crypto.addresses import AddressGenerator
-from iota.crypto.types import Seed
```
Comment (todofixthis) [inline examples/address_generator.py]: ❕ This import is used on line 60, as part of a type hint:

https://github.com/iotaledger/iota.lib.py/blob/develop/examples/address_generator.py#L59-L60

File: examples/send_transfer.py [hunk 2]
```diff
@@ -2,45 +2,115 @@
...
+    )
+
+    parser.add_argument(
+        '--seed',
+        type=text_type,
+        default=b'THESEEDOFTHEWALLETSENDINGGOESHERE99999999999999999999999999999999999999999999999',
+        help=
+        'Users IOTA seed.'
+        '(defaults to THESEEDOFTHEWALLETSENDINGGOESHERE99999999999999999999999999999999999999999999999).',
+    )
```
Comment (todofixthis) [inline examples/send_transfer.py]: ❔ I'm a tad bit uncomfortable having an interface for users to enter their seed in cleartext.  Could we change this to use the functions from https://github.com/iotaledger/iota.lib.py/blob/develop/examples/address_generator.py#L46-L80 instead?

File: examples/address_generator.py (hunk 1 above)
Comment (EmbeddedAndroid) [reply examples/address_generator.py]: Ah good catch, should have looked closed. Will drop this patch.

File: examples/send_transfer.py (hunk 2 above)
Comment (EmbeddedAndroid) [reply examples/send_transfer.py]: Travelling at the moment, but will implement this as soon as I can.

Comment (todofixthis) [reply examples/send_transfer.py]: Cheers mate!

**Comment by EmbeddedAndroid** [comment]
@todofixthis comments addressed finally :)

**Pull request merged**